scanpy/_metadata.py

```diff
@@ -19,3 +38,10 @@
     __version__ = metadata['Version']
     __author__ = metadata['Author']
     __email__ = metadata['Author-email']
+
+
+def within_flit():
+    for frame in traceback.extract_stack():
+        if frame.name == 'get_docstring_and_version_via_import':
+            return True
+    return False
```

Comment:
Why is this needed?

Reply:
Because otherwise, `__version__` can’t be retrieved without all dependencies being installed.

We talked about this quite some time ago. Some more discussion here: https://github.com/takluyver/flit/issues/253#issuecomment-737870438

I hope @takluyver gets around to merge & release takluyver/flit#382 one of these days, but this PR won’t wait on that.

Reply:
Can you document this in the code? Should have an explanation of why this exists, and a link to the pr so it's easy to know when we can remove it.

Reply:
sure!